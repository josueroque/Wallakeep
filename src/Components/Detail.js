import React,{Component,Fragment} from 'react';
import { getAd } from '../Api/Api';
import Header from './Header';
//import { Link } from "react-router-dom";
class Detail extends Component {
    constructor(props){
        super(props);
        const { adId,pathname } = this.props.location.state
        console.log(pathname);
        console.log('hola '+adId);
        this.state = { 
           _id:adId,
           Ad:[]
         };

    
    this.getAdvert(adId);
    }
    
    getAdvert=async(id)=>{
        const Advert=await getAd(id);
        this.setState({
            Ad:Advert,
            
        });
        console.log('Ad'+Advert['name']);
    }

    render() { 
        return (  
            <Fragment>
                <Header />
                <div className="card" >
                    <div className="card-image" >
                        <img  src={`http://localhost:3001/${this.state.Ad.photo}`} alt='Imagen de anuncio' >
                        </img>
                    </div>    
                    <h3>{this.state.Ad.name}</h3>
                </div> 
             </Fragment>
        );
    }
}
 
export default Detail;