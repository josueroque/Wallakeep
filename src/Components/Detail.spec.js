import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail';

describe.skip('Detail', () => {
  const defaultProps = {
       location: {
        pathname: "/detail/5db1de95f108b11d002b9e03",
        state: { adId: "5db1de95f108b11d002b9e03"} 
   }
};
  
  const render = props => shallow(<Detail {...defaultProps} {...props} />);
  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

//console.log(defaultProps);
  it.skip('should render the product detail', () => {
    expect(wrapper.find('.state').props().items).toHaveLength(0);
  });

});
