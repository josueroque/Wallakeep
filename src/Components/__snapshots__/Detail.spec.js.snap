// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail should render 1`] = `
<Fragment>
  <withRouter(Navbar) />
  <div
    className="detail-section"
  >
    <div
      className="card-image"
    />
    <div
      className="content"
    >
      <h3
        className="text-center"
      />
      <p />
      <p>
        Precio: $ undefined
      </p>
      <p>
        Tipo: undefined
      </p>
      <Link
        to={
          Object {
            "pathname": "/Create-Edit/undefined",
            "state": Object {
              "adId": undefined,
            },
          }
        }
      >
        <input
          className="uk-button uk-button-danger"
          type="submit"
          value="Editar"
        />
      </Link>
    </div>
  </div>
</Fragment>
`;
